File: src/main/java/org/railwaystations/rsapi/services/CountryService.java
package org.railwaystations.rsapi.services;

import org.railwaystations.rsapi.adapter.db.CountryDao;
import org.railwaystations.rsapi.domain.model.Country;
import org.springframework.stereotype.Service;

import java.util.Collection;

@Service
public class CountryService {

    private final CountryDao countryDao;

    public CountryService(final CountryDao countryDao) {
        this.countryDao = countryDao;
    }

    public Collection<Country> list(final Boolean onlyActive) {
        return countryDao.list(onlyActive == null || onlyActive);
    }

}

File: core/src/main/kotlin/org/railwaystations/rsapi/core/services/CountryService.kt
package org.railwaystations.rsapi.core.services

import org.railwaystations.rsapi.core.model.Country
import org.railwaystations.rsapi.core.ports.inbound.ListCountriesUseCase
import org.railwaystations.rsapi.core.ports.outbound.CountryPort
import org.springframework.stereotype.Service

@Service
class CountryService(private val countryDao: CountryPort) : ListCountriesUseCase {
    override fun list(onlyActive: Boolean?): Collection<Country> {
        return countryDao.list(onlyActive == null || onlyActive)
    }
}
